app/helpers/user_services_helper.rb

```diff
@@ -1,5 +1,84 @@
+# def get_create_user_params
+#   @input = parse_request
+#   print @input
+#   validates_presence_of :email_id, :password, :gender, :address
+# end
+
+def get_create_address_params
+  @input = parse_request
+  validates_presence_of :name, :line1, :line2, :city, :state, :pincode , :phone_number
+end
+
+def create_address(address_params, user_id)
+  address_params[:user_id] = get_valid_user_id user_id
+  address_params[:is_primary] = is_primary_address address_params
+  address = Address.transaction do
+    Address.create!(address_params)
+  end
+  return address
+end
+
+def get_update_address_params
+  @input = parse_request
+  validates_presence_of :address_id, :address
+end
+
+def get_delete_addresses_params
+  @input = parse_request
+  validates_presence_of :address_ids
+end
+
 def get_create_user_params
   @input = parse_request
-  print @input
-  validates_presence_of :email_id, :password, :gender, :address
+  validates_presence_of :email_id
+end
+
+def update_user_params
+  @user_params = {} #defining empty hash
+  @user_params[:email_id] = @input[:email_id]
+  if @input[:password].nil?
+    @user_params[:password] = 'opendoors'
```

Comment:
How will someone access anyone's info? Our access is solely through google. For login_type='Google', password doesnt make sense